sktime/transformations/panel/channel_selection.py

```diff
@@ -43,81 +37,189 @@ def _detect_knee_point(values, indices):
     knee = values[knee_idx]
     best_dims = [idx for (elem, idx) in zip(values, indices) if elem > knee]
     if len(best_dims) == 0:
-        return [knee_idx], knee_idx
+        # return all dimensions if no elbow point is found
+        return indices
+    return best_dims
 
-    return (best_dims,)
 
+def create_distance_matrix(
+    prototype: Union[pd.DataFrame, np.ndarray],
+    class_vals: np.array,
+    distance_: str = "euclidean",
+) -> pd.DataFrame:
+    """Create a distance matrix between class_prototypes.
 
-class _distance_matrix:
-    """Create distance matrix."""
+    Parameters
+    ----------
+    prototype : pd.DataFrame or np.ndarray
+        Prototype for each class.
+    class_vals : np.array
+        Class values.
+    distance_ : str, default="euclidean"
+        Distance metric to be used for calculating distance between class prototypes.
+
+    Returns
+    -------
+    distance_frame : pd.DataFrame
+        Distance matrix between class prototypes.
 
-    def distance(self, centroid_frame):
-        """Fuction to create DM."""
-        distance_pair = list(
-            itertools.combinations(range(0, centroid_frame.shape[0]), 2)
-        )
-        # exit()
-
-        map_cls = centroid_frame.class_vals.to_dict()
-        distance_frame = pd.DataFrame()
-        for class_ in distance_pair:
-
-            class_pair = []
-            # calculate the distance of centroid here
-            for _, (q, t) in enumerate(
-                zip(
-                    centroid_frame.drop(["class_vals"], axis=1).iloc[class_[0], :],
-                    centroid_frame.iloc[class_[1], :],
+    """
+    assert prototype.shape[0] == len(
+        class_vals
+    ), "Prototype and class values must be of same length."
```

Comment:
Assets are dangerous in production since they can be switched by an interpreter flag, consider changing to a if/except block
```suggestion
    if prototype.shape[0] != len(class_vals):
        raise ValueError(f"Prototype {prototype.shape[0]} and class values {len(class_vals)} must be of same length.")
```